scripts/zkai_stwo_query_grinding_budget_gate.py

```
@@ -0,0 +1,619 @@
+
+def payload_commitment(payload: dict[str, Any]) -> str:
+    item = copy.deepcopy(payload)
+    item.pop("payload_commitment", None)
+    return blake2b_commitment(PAYLOAD_DOMAIN, item)
+
+
+def parse_int(row: dict[str, str], field: str) -> int:
+    try:
+        return int(row[field])
+    except (KeyError, TypeError, ValueError) as err:
+        raise StwoQueryGrindingBudgetGateError(f"inventory field {field} is invalid") from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)